### PR TITLE
Breaking: Require ruby 2.3 or newer, to be able to use latest Cucumber, for 2.7 support

### PR DESCRIPTION
**Pull request by olleolleolle**
This PR updates the required_ruby_version to `>= 2.3`, in order to be able to run Cucumber tests.

- Add Ruby 2.7 to test matrix - Fixes #787
- Prepare for Ruby 2.7 compatibility
- Also: Update the `excon` library - Fixes #814
- Update cucumber to latest release
- Modernize Cucumber configuration
- Breaking: Drop Cucumber 1.3 support
- Breaking: Update the required_ruby_version to `>= 2.3`

### CODE REVIEW

**Comment by olleolleolle** [comment]
Cc @mvz

**Comment by olleolleolle** [comment]
**Update**: Learning that the "features/test_frameworks/cucumber_feature" does not currently find more than 0 tests. Investigating.

File: spec/lib/vcr/cassette/serializers_spec.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@ class Cassette
               result = serializer.serialize("a" => string)
               serializer.deserialize(result)
             }.to raise_error(error_class, /preserve_exact_body_bytes/)
-          end unless (RUBY_INTERPRETER == :rubinius && RUBY_VERSION =~ /^1.9/)
+          end unless (RUBY_INTERPRETER == :rubinius)
```
Comment (mvz) [inline spec/lib/vcr/cassette/serializers_spec.rb:20]: I think the original meant that the spec was not skipped on Rubinius targeting Ruby 2.0 and up. So the whole conditional can be removed.

**Comment by choosen** [comment]
Can you mention explicit about support ruby 2.7? 
I like that there is info about 2.3 >=   but explicit 2.7 would be great

**Comment by olleolleolle** [comment]
@choosen I'd merge a small PR if you have one?

**Pull request merged**